android/src/main/kotlin/com/adora/flutter_notification_listener_service/FlutterNotificationListenerServicePlugin.kt

```diff
@@ -1,35 +1,205 @@
+                }
+            }
+            return isPermissionGranted
+        }
+
+        private fun enableServiceSettings(result: MethodChannel.Result){
+            if(context==null){
+                result.error("permission-error","Unable to enable service settings","context not mounted")
+                return
+            }
+            val receiver = ComponentName(context!!, NotificationsHandlerService::class.java)
+            val pm = context!!.packageManager
+            pm.setComponentEnabledSetting(receiver, PackageManager.COMPONENT_ENABLED_STATE_ENABLED, PackageManager.DONT_KILL_APP)
+            result.success(true)
+            return
+        }
+
+        private fun disableServiceSettings(result: MethodChannel.Result){
+            if(context==null){
+                result.error("permission-error","Unable to request Permission","context not mounted")
+                return
+            }
+            context!!.startActivity(Intent(ACTION_NOTIFICATION_LISTENER_SETTINGS).addFlags(Intent.FLAG_ACTIVITY_NEW_TASK))
+            result.success(true)
+            return
+        }
+
+        private fun startService(result: MethodChannel.Result){
+            if(context==null){
+                result.error("permission-error","Unable to start service","context not mounted")
+                return
+            }
+            if(!permissionStatus()){
+                result.error("permission-missing","Unable to start the service as the permission is missing",null)
+                return
+            }
+
+
+            /// Set up listener intent
+            if(listenerIntent!=null) {
+                result.error("service-already-running","The service is already running. Please stop and start again",null)
+                return
+            }
+            listenerIntent = Intent(
+                context,
+                NotificationsHandlerService::class.java
+            )
+
+            context!!.startService(listenerIntent)
+
+            Log.i(tag, "Started the native NotificationListenerService.")
+            result.success(true)
+            return
+        }
+
+        private fun stopService(result: MethodChannel.Result){
+            if(context==null){
+                result.error("permission-error","Unable to stop service","context not mounted")
+                return
+            }
+            if(listenerIntent!=null) {
+                context!!.stopService(listenerIntent)
+                listenerIntent = null
+                Log.i(tag, "Stopped the native NotificationListenerService.")
+                result.success(true)
+                return
+            }
+            Log.i(tag, "Native NotificationListenerService was not running when stop was called")
+            result.success(true)
+            return
+        }
+
+      override fun onDetachedFromEngine(binding: FlutterPluginBinding) {
+        channel!!.setMethodCallHandler(null)
+        eventChannel!!.setStreamHandler(null)
```

Comment:
```suggestion
        channel?.setMethodCallHandler(null)
        eventChannel?.setStreamHandler(null)
```
non-nullabilty isn't guaranteed. Service could start and stop arbitrarily iirc